src/main/UserActivityMonitor.js

```diff
@@ -0,0 +1,170 @@
+   *
+   * @param {Object} config - overide internal configuration defaults
+   * @param {nunber} config.regularUpdateFrequencyMs
+   * @param {nunber} config.statusUpdateThresholdMs
+   * @param {nunber} config.userActivityTimeoutSec
+   * @emits {error} emitted when method is called before the app is ready
+   * @emits {error} emitted when this method has previously been called but not subsequently stopped
+   */
+  startMonitoring(config = {}) {
+    if (!app.isReady()) {
+      this.emit('error', new Error('UserActivityMonitor.startMonitoring can only be called after app is ready'));
+      return;
+    }
+
+    if (this.systemIdleTimeIntervalID >= 0) {
+      this.emit('error', new Error('User activity monitoring is already in progress'));
+      return;
+    }
+
+    this.config = Object.assign({}, this.config, config);
+
+    // NOTE: electron.powerMonitor cannot be referenced until the app is ready
+    electron.powerMonitor.on('suspend', this.handleSuspend);
+    electron.powerMonitor.on('resume', this.handleResume);
+    electron.powerMonitor.on('shutdown', this.handleShutdown);
+    electron.powerMonitor.on('lock-screen', this.handleLockScreen);
+    electron.powerMonitor.on('unlock-screen', this.handleUnlockScreen);
+
+    this.systemIdleTimeIntervalID = setInterval(() => {
+      try {
+        electron.powerMonitor.querySystemIdleTime((idleTime) => {
+          this.updateIdleTime(idleTime);
+        });
+      } catch (err) {
+        console.log('Error getting system idle time:', err);
+      }
+    }, this.config.regularUpdateFrequencyMs);
+  }
+
+  /**
+   * Stop monitoring system events and idle time
+   */
+  stopMonitoring() {
+    electron.powerMonitor.off('suspend', this.handleSuspend);
+    electron.powerMonitor.off('resume', this.handleResume);
+    electron.powerMonitor.off('shutdown', this.handleShutdown);
+    electron.powerMonitor.off('lock-screen', this.handleLockScreen);
+    electron.powerMonitor.off('unlock-screen', this.handleUnlockScreen);
+
+    clearInterval(this.systemIdleTimeIntervalID);
+  }
+
+  /**
+   * Updates internal idle time properties and conditionally triggers updates to user activity status
+   *
+   * @param {integer} idleTime
+   * @private
+   */
+  updateIdleTime(idleTime) {
+    this.idleTime = idleTime;
+
+    if (this.idleTime > this.config.userActivityTimeoutSec) {
+      this.updateUserActivityStatus(false);
+    } else if (!this.forceUserIsInactive && this.idleTime < this.config.userActivityTimeoutSec) {
+      this.updateUserActivityStatus(true);
+    }
+  }
+
+  /**
+   * Updates user activity status if changed and triggers a status update
+   *
+   * @param {boolean} isActive
+   * @private
+   */
+  updateUserActivityStatus(isActive = false) {
+    if (isActive !== this.isActive) {
+      this.isActive = isActive;
+      this.sendStatusUpdate();
+    } else if (new Date().getTime() - this.lastStatusUpdate > this.config.statusUpdateThresholdMs) {
```

Comment:
`new Date().getTime()` could be shortened to `Date.now()`.

Also, is it possible that the "now" time here will be different than the "now" time in `sendStatusUpdate`? Perhaps we should get that time once and pass it as an argument to `sendStatusUpdate`. 0/5 on that though since I think it's a good practice, but I don't see any real issues that could result from it.